VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTopic.java

```diff
@@ -80,7 +83,9 @@ public boolean hasSameId(@NonNull DiscussionTopic discussionTopic) {
     }
 
     public boolean containsThread(@NonNull DiscussionThread discussionThread) {
-        if (identifier.equals(discussionThread.getTopicId())) {
+        if (isAllType() || isFollowingType()) {
```

Comment:
I don't think it will have a noticable impact now, but for correctness sake, could this be `(isFollowingType() && discussionThread.isFollowing())`?